### PR TITLE
Move fallback gradient output to the end

### LINKED ISSUE #8: Incompatible with complex background shorthand
Let's say I have this code:

```css
background: resemble-image(url("foo.png")) no-repeat 75% center / cover;
```

This would be generated into:

```css
background: url("foo.png"), linear-gradient(90deg, #161e2a 0%, #1a2331 25%, #5c4e52 50%, #584b4f 75%) no-repeat 75% center / cover;
```

Which would invalidate it.

Now I can think of either making the parser work by detecting commas to know when to put the fallback in (or just whack it at the end of the value with a `,` prefix, or wrap the whole long shorthand value into the `resemble-image` function itself. I guess this would be your call on how to proceed 😄

### PR DESCRIPTION
**Pull request by diagramatics**
Fixes #8. Thanks for sorting out what to do to fix it really quickly!

I've also added a test, but we might need a test to check if the gradient is added at the end. I've also added another test to check if `resemble-image()` is removed in the output.

### CODE REVIEW

File: src/__tests__/index.js [hunk 1]
```diff
@@ -152,6 +165,12 @@ test(
     4
 );
 
+test('should handle background value shorthand',
+    assertColourStops,
+    `header{background:resemble-image(url("${image}")) no-repeat center / cover}`,
```
Comment (ben-eb) [inline src/__tests__/index.js:170]: Perhaps we want to test that the gradient makes it to the end of the value? Something like this:

```js
test('should handle background value shorthand', t => {
  const fixture = `header{background:resemble-image(url("${image}")) no-repeat center / cover}`;
  return postcss(plugin()).process(fixture).then((result) => {
    t.is(result.root.first.nodes[0].value.indexOf('linear-gradient'), 30); // replace with the actual index!
    return assertColourStops(t, fixture, 4);
  });
});

**Review by ben-eb — CHANGES_REQUESTED** [review]

File: src/__tests__/index.js (hunk 1 above)
Comment (diagramatics) [reply src/__tests__/index.js:170]: Seems like a regex is a more robust choice if we don't want to hardcode the index in the test. Updated with the next commit.

**Review by ben-eb — APPROVED** [review]

**Comment by ben-eb** [comment]
Many thanks for your contribution! I've released this now as `2.1.1`. 👍

**Comment by diagramatics** [comment]
Perfect! Looking forward to using this in a production site 🎉

**Pull request merged**